src/main/java/greencity/service/impl/PlaceServiceImpl.java

```diff
@@ -78,87 +76,100 @@ public Place save(PlaceAddDto dto, String email) {
         Category category = categoryService.findByName(dto.getCategory().getName());
         Place place = modelMapper.map(dto, Place.class);
 
-        place.setAuthor(userService.findByEmail(email).orElseThrow(
-            () -> new NotFoundException(ErrorMessage.USER_NOT_FOUND_BY_EMAIL)));
+        getUserByEmailAndSetToPlace(email, place);
+        place.setCategory(category);
+        place.setLocation(modelMapper.map(dto.getLocation(), Location.class));
```

Comment:
so the location is saving now when you save place?
but why can't you use the same approach to Discounts and Opening hours so that you use separate "save"?